Instastories.py

```diff
@@ -44,6 +46,32 @@ def get_media(url, path, media_type, username):
         base64_media = (base64.b64encode(image.read()).decode("utf-8"), media_type, username)
         return base64_media
 
+def save_cached_id(username, new_cached_id):
+    cached_ids = get_cached_ids()
+    cached_ids[username] = new_cached_id
+    with open(DEFAULT_CACHED_IDS_PATH, "w+") as file:
+        json.dump(cached_ids, file)
```

Comment:
You are calling this function for every id. 
This mean you open close the file 2 times each ids, not great. Store them in memory and load/save only once.